PatientWebApplication/HealthClinic.DAL/Service/OperationService.cs

```diff
@@ -55,7 +58,68 @@ public Boolean isTermNotAvailable(DoctorUser doctor, TimeSpan start, TimeSpan en
+        public List<Operation> SimpleSearchOperations(AppointmentReportSearchDto appointmentReportSearchDto)
+        {
+            List<Operation> operations = GetOperationsForPatient(appointmentReportSearchDto.PatientId);
+
+            operations = SearchForDate(operations, appointmentReportSearchDto);
+
+            operations = SearchForDoctorNameAndSurname(operations, appointmentReportSearchDto);
+
+            operations = SearchForAppointmentType(operations, appointmentReportSearchDto);
+
+            return operations;
+
+        }
+
+        private List<Operation> SearchForDoctorNameAndSurname(List<Operation> operations, AppointmentReportSearchDto appointmentSearchDto)
+        {
+            if (!appointmentSearchDto.DoctorNameAndSurname.Equals(""))
+            {
+                operations = operations.FindAll(operation => operation.Doctor.firstName.Contains(appointmentSearchDto.DoctorNameAndSurname) || operation.Doctor.secondName.Contains(appointmentSearchDto.DoctorNameAndSurname));
+            }
+
+            return operations;
+        }
+
+        private List<Operation> SearchForDate(List<Operation> operations, AppointmentReportSearchDto appointmentSearchDto)
+        {
+            if (!appointmentSearchDto.Start.Equals("") && !appointmentSearchDto.End.Equals(""))
+            {
+                DateTime startDate = DateTime.ParseExact(appointmentSearchDto.Start, "dd/MM/yyyy", CultureInfo.InvariantCulture);
+                DateTime endDate = DateTime.ParseExact(appointmentSearchDto.End, "dd/MM/yyyy", CultureInfo.InvariantCulture);
+                operations = operations.FindAll(operation => startDate <= DateTime.ParseExact(operation.Date, "dd/MM/yyyy", CultureInfo.InvariantCulture) && DateTime.ParseExact(operation.Date, "dd/MM/yyyy", CultureInfo.InvariantCulture) <= endDate);
+            }
+            else if (appointmentSearchDto.Start.Equals("") && !appointmentSearchDto.End.Equals(""))
+            {
+                DateTime endDate = DateTime.ParseExact(appointmentSearchDto.End, "dd/MM/yyyy", CultureInfo.InvariantCulture);
+                operations = operations.FindAll(operation => DateTime.ParseExact(operation.Date, "dd/MM/yyyy", CultureInfo.InvariantCulture) <= endDate);
+            }
+            else if (!appointmentSearchDto.Start.Equals("") && appointmentSearchDto.End.Equals(""))
+            {
+                DateTime startDate = DateTime.ParseExact(appointmentSearchDto.Start, "dd/MM/yyyy", CultureInfo.InvariantCulture);
+                operations = operations.FindAll(operation => startDate <= DateTime.ParseExact(operation.Date, "dd/MM/yyyy", CultureInfo.InvariantCulture));
+            }
+
+            return operations;
+
+        }
+
+        private List<Operation> SearchForAppointmentType(List<Operation> operations, AppointmentReportSearchDto appointmentSearchDto)
+        {
+            if (appointmentSearchDto.AppointmentType.Equals("") || appointmentSearchDto.AppointmentType.Equals("Operation"))
```

Comment:
Apply a method that checks to see if it is an empty string. Create a method that checks if the appointment type is an operation and apply it in the if statement.